### PR TITLE
Combine collection and token under the digitalAsset namespace

### PR DESCRIPTION
**Pull request by heliuchuan**
### Test Plan
pnpm test

### CODE REVIEW

File: src/internal/digitalAsset.ts [hunk 1]
```diff
@@ -181,3 +196,131 @@ export async function getTokenActivity(args: {
...
+        new Bool(args.mutableRoyalty ?? true),
+        new Bool(args.mutableURI ?? true),
+        new Bool(args.mutableTokenDescription ?? true),
+        new Bool(args.mutableTokenName ?? true),
+        new Bool(args.mutableTokenProperties ?? true),
+        new Bool(args.mutableTokenURI ?? true),
+        new Bool(args.tokensBurnableByCreator ?? true),
+        new Bool(args.tokensFreezableByCreator ?? true),
+        new U64(args.royaltyNumerator ?? 0),
+        new U64(args.royaltyDenominator ?? 1),
```
Comment (0xmaayan) [inline src/internal/digitalAsset.ts:247]: that looks pretty damn good 
cc @xbtmatt

File: src/internal/digitalAsset.ts [hunk 2]
```diff
@@ -181,3 +196,131 @@ export async function getTokenActivity(args: {
...
+ * @param creatorAddress the address of the collection's creator
+ * @param collectionName the name of the collection
+ * @returns GetCollectionDataResponse response type
+ */
+export async function getCollectionData(args: {
+  aptosConfig: AptosConfig;
+  creatorAddress: HexInput;
+  collectionName: string;
+  options?: {
+    tokenStandard?: TokenStandard;
```
Comment (0xmaayan) [inline src/internal/digitalAsset.ts:270]: I wish we could just ignore it and support only v2

File: src/internal/digitalAsset.ts [hunk 3]
```diff
@@ -181,3 +196,131 @@ export async function getTokenActivity(args: {
...
+    query: graphqlQuery,
+    originMethod: "getCollectionData",
+  });
+
+  if (data.current_collections_v2.length === 0) {
+    throw Error("Collection not found");
+  }
+
+  if (data.current_collections_v2.length > 1) {
+    throw Error(
```
Comment (0xmaayan) [inline src/internal/digitalAsset.ts:302]: not sure if its an error...

File: tests/e2e/api/digitalAsset.test.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  return (
+    await aptos.getOwnedTokens({
+      ownerAddress: creator.accountAddress.toString(),
+    })
+  )[0].current_token_data?.token_data_id!;
+}
+
+describe("DigitalAsset", () => {
+  let tokenAddress: string;
+  //   let collectionAddress
```
Comment (0xmaayan) [inline tests/e2e/api/digitalAsset.test.ts:55]: remove?

File: src/internal/digitalAsset.ts [hunk 5]
```diff
@@ -3,29 +3,44 @@
...
- * the {@link api/token}. By moving the methods out into a separate file,
+ * the {@link api/digitalAsset}. By moving the methods out into a separate file,
  * other namespaces and processes can access these methods without depending on the entire
- * token namespace and without having a dependency cycle error.
+ * digitalAsset namespace and without having a dependency cycle error.
  */
 
 import { AptosConfig } from "../api/aptosConfig";
-import { MoveString, MoveVector, U8 } from "../bcs";
+import { Bool, U64, U8 } from "../bcs/serializable/movePrimitives";
```
[the remark this answers is not in the document]
Comment (heliuchuan) [reply src/internal/digitalAsset.ts:12]: no i just let vscode auto-import, wish it was smarter...
fixed!

File: src/internal/digitalAsset.ts (hunk 3 above)
Comment (heliuchuan) [reply src/internal/digitalAsset.ts:302]: defaulted token standard to v2

File: src/internal/digitalAsset.ts (hunk 2 above)
Comment (heliuchuan) [reply src/internal/digitalAsset.ts:270]: defaulted the arg to v2 at least

File: src/internal/digitalAsset.ts (hunk 1 above)
Comment (xbtmatt) [reply src/internal/digitalAsset.ts:246]: :D I know right! I know some people like the remote ABI but I love seeing the types

**Pull request merged**